admin-dev/themes/new-theme/js/pages/order/create/addresses-renderer.js

```diff
@@ -36,68 +36,74 @@ export default class AddressesRenderer {
    * @param {Array} addresses
    */
   render(addresses) {
-    let deliveryAddressDetailsContent = '';
-    let invoiceAddressDetailsContent = '';
-
-    const $deliveryAddressDetails = $(createOrderPageMap.deliveryAddressDetails);
-    const $invoiceAddressDetails = $(createOrderPageMap.invoiceAddressDetails);
-    const $deliveryAddressSelect = $(createOrderPageMap.deliveryAddressSelect);
-    const $invoiceAddressSelect = $(createOrderPageMap.invoiceAddressSelect);
-
-    const $addressesContent = $(createOrderPageMap.addressesContent);
-    const $addressesWarningContent = $(createOrderPageMap.addressesWarning);
-
-    $deliveryAddressDetails.empty();
-    $invoiceAddressDetails.empty();
-    $deliveryAddressSelect.empty();
-    $invoiceAddressSelect.empty();
-
+    this._cleanAddresses();
     if (addresses.length === 0) {
-      $addressesWarningContent.removeClass('d-none');
-      $addressesContent.addClass('d-none');
+      this._hideAddressesContent();
+      this._showEmptyAddressesWarning();
+      this._showAddressesBlock();
 
       return;
     }
 
-    $addressesContent.removeClass('d-none');
-    $addressesWarningContent.addClass('d-none');
+    this._showAddressesContent();
+    this._hideEmptyAddressesWarning();
 
-    for (const key in Object.keys(addresses)) {
+    for (const key in addresses) {
       const address = addresses[key];
 
-      const deliveryAddressOption = {
-        value: address.addressId,
-        text: address.alias,
-      };
-
-      const invoiceAddressOption = {
-        value: address.addressId,
-        text: address.alias,
-      };
-
-      if (address.delivery) {
-        deliveryAddressDetailsContent = address.formattedAddress;
-        deliveryAddressOption.selected = 'selected';
-      }
+      this._renderDeliveryAddress(address);
+      this._renderInvoiceAddress(address);
+    }
 
-      if (address.invoice) {
-        invoiceAddressDetailsContent = address.formattedAddress;
-        invoiceAddressOption.selected = 'selected';
-      }
+    // @todo: router.generate(admin_addresses_create) when pr #15300 merged
```

Comment:
We are not 100% sure we'll be able to merge PR #15300 for 1.7.7 feature freeze 😥 so careful, we might have to rely on legacy page here. Will this be an issue ?

Reply:
This shouldn't be hard to replace with legacy, but the addresses_edit route might be.